### DISCUSSION TITLE
Help with addressing Koinly .csv errors

### DISCUSSION THREAD

**Discussion by overdrive6** [pwillworth/dfkreport#11] [Q&A] [answered]
Hi there,

Sorry as I am new to handling crypto taxes.  First off, thank you for putting together this tool to help with keeping track of taxes.

I added my Harmony and DFK Chain wallets and generated a .csv for koinly and I got the following errors.  Can someone help with resolving them? For the "1USDC" error, I resolved that by changing it to USDC and reuploading the .csv to koinly.  I am not sure how to handle the rest of them, like LP token (and toekn?), xJEWEL, and airdrops/forks.

<img width="1155" alt="Screen Shot 2022-04-07 at 11 24 44 AM" src="https://user-images.githubusercontent.com/103217409/162282870-61a233dc-a69e-47b5-ac72-38de526b455a.png">
<img width="1137" alt="Screen Shot 2022-04-07 at 11 24 50 AM" src="https://user-images.githubusercontent.com/103217409/162282886-623c3e60-57cb-4a05-84cb-6b7da7f00660.png">
<img width="758" alt="Screen Shot 2022-04-07 at 11 32 44 AM" src="https://user-images.githubusercontent.com/103217409/162282903-6e4540c0-86b3-4375-8b4d-7c8c9035bdf6.png">

Thank you in advance!

**Comment by pwillworth** [comment, marked as answer]
Hi there!  It is really up to you how you want to handle those errors.  All I can say is there is a lot of stuff that is happening in the game that Koinly doesn't understand.  You can either modify the tokens to be something close enough to what koinly understands like you did with 1USDC, or remove the stuff that koinly doesn't understand and decide if it needs to be reported in some other way or not.

This gets into the grey areas of crypto taxes, and I am not a tax advisor.  This tool just reports what transactions occur in the game, but it is kinda up to you and/or your tax advisor on how to apply them.

**Reply by overdrive6** [reply]
Thank you for the response.  I'm learning so much about taxes and how it can be interpreted in many ways...it's not totally cut n dry.

I know you aren't an advisor but how did you handle the airdrops?  

I think it makes sense to relabel xJewel as Jewel...what do you think?

**Reply by pwillworth** [reply]
Personally I didn't have any airdrops last year, so didn't have to worry about it.  For the purposes of the tool, I just list them in the income section.  I don't know about the xJewel thing, I guess it depends on what the transaction is, you might want to convert the value to the jewel equivalent if you are going to change the label to Jewel.

**Reply by Sc1entifik** [reply]
I am having the same kind of issues and would like to point out some troublesome details this answer DID NOT address.

Firstly using the Koinly Universal csv format when dlding the file gave me the same errors of currencies not being found. Why is this so if I am downloading a file from your website marked for Koinly Universal?

Another issue is the fact that the only currency Koinly did recognize when I imported the file Crystal wasn't even the correct currency and was some other Crystal from some other unrelated project/blockchain!

If anyone on here can tell me how I they dealt with these issues using  these horrible tax tools like Koinly I would surely appreciate it.

**Reply by overdrive6** [reply]
I believe that DFKreport is a good site and it's a great tool to organize all of the DFK transactions for your taxes.  I am appreciative of  DFKreport (thank you developer(s))  

The issue is that Koinly (and other tax softwares) doesn't have all of the tokens in their system.  DeFi is moving so fast that tax software/websites can't keep up with everything. So it is up to us to understand this and adjust the files so that the proper information is reported for taxes.  Meaning we need to manually adjust the files so that the proper information is reported.

Good find about Crystal token.  I am going to take a look at this.
